Extension/src/LanguageServer/client.ts

```diff
@@ -179,45 +266,328 @@ function publishDiagnostics(params: PublishDiagnosticsParams): void {
     clientCollection.timeTelemetryCollector.setUpdateRangeTime(realUri);
 }
 
-function publishCodeAnalysisDiagnostics(params: PublishDiagnosticsParams): void {
+// Rebuild codeAnalysisCodeToFixes and codeAnalysisAllFixes.fixAllCodeActions.
+function rebuildCodeAnalysisCodeAndAllFixes(): void {
```

Comment:
Maybe it would make sense to move some of this code into a new `codeAnalysis.ts` file?

Reply:
Potentially -- but I didn't want to move existing code analysis code because it would make the PR hard to diff.

I'm definitely in favor of moving more stuff out of client.ts.

Should I do that in this PR or a follow up one?

Reply:
A later PR seems fine.

Reply:
I had started refactoring to a codeAnalysis.ts, but there were too many changes involved so I abandoned that (since the diff would be harder to see).

Reply:
Yeah, it would be easier to follow changes of a refactor in a separate PR. I would refactor first before adding/fixing features.

Reply:
Yeah, that sounds good. If I find any high priority issues that should be fixed I can do them in PR separate from the refactor and deal with any potential merge conflicts, depending on how long the refactor takes, etc.